scripts/benchmark/run.py

```diff
@@ -0,0 +1,275 @@
+    else:
+        raise Exception(f"No function found by name: {function_name}")
+
+
+def do_basic_benchmarks():
+    runtime = Runtime()
+    module, _ = runtime.instantiate_module(runtime.load_module(BASIC_WASM))
+    runtime.register_module('bench', module)
+
+    thunk_delta = _run_basic_bench(runtime, module, 'thunk', (), 10000)
+    add_delta = _run_basic_bench(runtime, module, 'add', (numpy.uint32(1), numpy.uint32(2)), 10000)
+    call_thunk_delta = _run_basic_bench(runtime, module, 'call_thunk', (numpy.uint32(10000),), 1)
+    call_add_delta = _run_basic_bench(
+        runtime, module, 'call_add', (numpy.uint32(10000), numpy.uint32(1), numpy.uint32(2)), 1,
+    )
+
+    logging.info(f"wrapped thunk: {call_thunk_delta / 10000:.8f} per call")
+    logging.info(f"wrapped add  : {call_add_delta / 10000:.8f} per call")
+
+    thunk_overhead = (call_thunk_delta - thunk_delta)
+    add_overhead = call_add_delta - add_delta
+    logging.info(f"call overhead for thunk: {thunk_overhead:.4f} ({100 * thunk_overhead / thunk_delta:.2f}%)  |   {thunk_overhead / 10000:.8f} per call")  # noqa: E501
+    logging.info(f"call overhead for add  : {add_overhead:.4f} ({100 * add_overhead / add_delta:.2f}%)  |   {add_overhead / 10000:.8f} per call")  # noqa: E501
+
+
+def _run_basic_bench(runtime, module, fn_name, fn_args, times):
+    signature = f"{fn_name}{fn_args}"
+    logging.info(f"BENCHMARKING: {signature}")
+    delta = time_fn(runtime, module, fn_name, fn_args, times)
+    logging.info(f"{signature.ljust(22)} x {str(times).rjust(10)}: {delta:.4f}  |  {delta / times:.4f} per-call")  # noqa: E501
+    return delta
+
+
+#
+# Tests from https://github.com/fluencelabs/fluence/tree/master/bench/vm/tests
+#
+def do_factorization_benchmarks(prime):
+    runtime = Runtime()
+
+    if prime == 1117:
+        module, _ = runtime.instantiate_module(runtime.load_module(FACTORIZATION_1117_WASM))
+    elif prime == 9973:
+        module, _ = runtime.instantiate_module(runtime.load_module(FACTORIZATION_9973_WASM))
+    elif prime == 2147483647:
+        module, _ = runtime.instantiate_module(runtime.load_module(FACTORIZATION_2147483647_WASM))
+    else:
+        raise Exception("Only 1117, 9973, 2147483647 are supported")
+
+    logging.info(f"BENCHMARKING: factorization of {prime}")
+    delta = time_fn(runtime, module, 'main', (), 1)
+    logging.info(f"took: {delta:.4f}")
+
+
+def do_fibonacci_bigint_benchmarks(order):
+    runtime = Runtime()
+
+    if order == 1:
+        module, _ = runtime.instantiate_module(runtime.load_module(FIBONACCI_1_WASM))
+    elif order == 11:
+        module, _ = runtime.instantiate_module(runtime.load_module(FIBONACCI_11_WASM))
+    elif order == 17:
+        module, _ = runtime.instantiate_module(runtime.load_module(FIBONACCI_17_WASM))
+    elif order == 38:
+        module, _ = runtime.instantiate_module(runtime.load_module(FIBONACCI_38_WASM))
+    else:
+        raise Exception("Only 1, 11, 17, 38 are supported")
```

Comment:
I'm 98% on board with how much explicit validation we do, but every once in a while I think we might go a bit overboard. This is one of those where I think: "how much worse would it be to dynamically do this and raise on failure?" like:

```py
def load_fib(order):
  fib_file = BASE_DIR / 'fibonacci_{order}_bigint_wasm.wasm'
  if not fib_file.exists():
    raise Exception("fibonacci {order} is not supported: {fib_file} not found")
  module = runtime.load_module(fib_file)
  return runtime.instantiate_module(module)
```
Then `do_fibonacci_bigint_benchmarks` gets simpler. Anyway, just a nitpick that pops in my head occasionally.

Reply:
:+1: at this point a lot of this is reflex/habit/well-worn-ruts for me.  Always happy for pushback as I get blind to this sometimes.  will simplify.